keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/binary/MatrixMultiplicationVertexTest.java

```diff
@@ -39,7 +39,7 @@ public void canDoMatrixMultiply2x2() {
         UniformVertex b = new UniformVertex(0, 10);
         b.setValue(DoubleTensor.create(new double[]{5, 6, 7, 8}, 2, 2));
 
-        MatrixMultiplicationVertex c = a.matrixMultiply(b);
+        MatrixMultiplicationVertex c = (MatrixMultiplicationVertex) a.matrixMultiply(b);
```

Comment:
Do you need this cast? Can it return something that doesn't need to be cast?

Reply:
You do if you explicitly call `.forwardModeAutoDiff`, so that the parameter types match.